src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -32,19 +32,29 @@ public void TestEnvironmentProperty()
 
             IDictionary<string, string> environment = psi.Environment;
 
-            Assert.NotEqual(environment.Count, 0);
+            Assert.NotEqual(0, environment.Count);
 
-            int CountItems = environment.Count;
+            int countItems = environment.Count;
 
             environment.Add("NewKey", "NewValue");
             environment.Add("NewKey2", "NewValue2");
 
-            Assert.Equal(CountItems + 2, environment.Count);
+            Assert.Equal(countItems + 2, environment.Count);
             environment.Remove("NewKey");
-            Assert.Equal(CountItems + 1, environment.Count);
+            Assert.Equal(countItems + 1, environment.Count);
 
-            //Exception not thrown with invalid key
-            Assert.Throws<ArgumentException>(() => { environment.Add("NewKey2", "NewValue2"); });
+            // The full .NET framework represents Environment with StringDictionary. On .NET Core we use
+            // use Dictionary<string, string>, because we need to have case sensitive behaviour on Unix.
```

Comment:
From @safern 

> @hughbe I believe this behavior was ported like this by design. I think it makes the implementation easier and better to understand. This was ported in #13203 by myself.

cc: @stephentoub @danmosemsft any thoughts?

I think we should open an issue as netfx-port-consider to port this to behave as core using dictionary instead of hash table.